client/dev-tools-promo/index.tsx

```diff
@@ -1,15 +1,30 @@
+import { FEATURE_SFTP, WPCOM_FEATURES_ATOMIC } from '@automattic/calypso-products';
 import page, { Context as PageJSContext } from '@automattic/calypso-router';
 import { makeLayout, render as clientRender } from 'calypso/controller';
 import { navigation, siteSelection, sites } from 'calypso/my-sites/controller';
 import { siteDashboard } from 'calypso/sites-dashboard-v2/controller';
 import { DOTCOM_DEVELOPER_TOOLS_PROMO } from 'calypso/sites-dashboard-v2/site-preview-pane/constants';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import { devToolsPromo } from './controller';
 
 const redirectForNonSimpleSite = ( context: PageJSContext, next: () => void ) => {
 	const state = context.store.getState();
 	const site = getSelectedSite( state );
-	if ( site && site.jetpack && ! site.plan?.expired ) {
+	const siteId = site?.ID ?? null;
+	const isJetpackSite = !! site?.jetpack;
+	const hasAtomicFeature = siteHasFeature( state, siteId, WPCOM_FEATURES_ATOMIC );
+	const hasSftpFeature = siteHasFeature( state, siteId, FEATURE_SFTP );
+	const isPlanExpired = !! site?.plan?.expired;
+
+	// Check for site features to determine if the promo should be shown in case the site has yet to transfer to Atomic.
+	const shouldShowDevToolsPromo =
+		! isJetpackSite &&
+		( isPlanExpired ||
+			! hasAtomicFeature ||
+			( ! hasSftpFeature && ! site?.is_wpcom_staging_site ) );
+
+	if ( ! shouldShowDevToolsPromo ) {
```

Comment:
my observation:
- The `Dev Tools` tab is hidden when I select a site ✅
- However, it seems to me that if I directly go to `/dev-tools-promo/<site>`, the `Dev Tools` tab is shown and it is not redirected. From here, if I select the `Overview` tab, the `Dev Tools` tab is gone.